vision_evaluation/evaluators.py

```diff
@@ -1157,3 +1157,54 @@ def add_predictions(self, predictions, targets):
 
     def get_report(self, **kwargs):
         return {f'{self._get_id()}': (float(self.total_error)**(1 / self.p) / self.total_num) if self.total_num else 0.0}
+
+
+class ConfusionMatrixEvaluator(Evaluator):
+    """
+    Confusion matrix evaluator
+    """
+    def reset(self):
+        super(ConfusionMatrixEvaluator, self).reset()
+        self.all_targets = np.array([])
+        self.all_predictions = np.array([])
+
+    def _get_id(self):
+        return 'confusion_matrix'
+
+    def add_predictions(self, predictions, targets):
+        """ Evaluate a batch of predictions and targets.
+        Args:
+            predictions: the model output numpy array. Shape (N,)
+            targets: the ground truth labels. Shape (N,)
+        """
+        assert predictions.shape == targets.shape
+        assert len(targets.shape) == 1
+
+        self.all_predictions = np.append(self.all_predictions, predictions, axis=0)
+        self.all_targets = np.append(self.all_targets, targets, axis=0)
+
+    def get_report(self, **kwargs):
+        """Get the calculated value of the metric
+        Args:
+            labels: List of labels to index the matrix. Array-like of shape (n_classes)
+            normalize: Boolean flag. If True, it normalizes the returned confusion matrix is normalized by the sum of the rows.
+        Returns:
+            returns a dict containing 2 entries:
```

Comment:
actually the only dict key is `confusion_matrix`, second layer is "cm" and "labels"